cw-orch-daemon/src/senders/manual_sender.rs

```diff
@@ -0,0 +1,178 @@
+use crate::proto::injective::InjectiveEthAccount;
+use crate::queriers::Node;
+use crate::tx_broadcaster::assert_broadcast_code_cosm_response;
+use crate::{cosmos_modules, DaemonBase, TxBuilder};
+
+use crate::{error::DaemonError, tx_resp::CosmTxResponse};
+
+use cosmrs::proto::cosmos;
+use cosmrs::proto::cosmos::auth::v1beta1::BaseAccount;
+use cosmrs::proto::cosmos::vesting::v1beta1::PeriodicVestingAccount;
+use cosmrs::tendermint::chain::Id;
+use cosmrs::tx::{ModeInfo, Raw, SignDoc, SignMode, SignerInfo};
+use cosmrs::{AccountId, Any};
+use cosmwasm_std::Addr;
+use cw_orch_core::environment::ChainInfoOwned;
+use prost::Message;
+use tonic::transport::Channel;
+
+use std::io::{self, BufRead};
+
+use super::sender_trait::SenderTrait;
+
+pub type ManualDaemon = DaemonBase<ManualSender>;
+
+#[derive(Clone, Default)]
+pub struct ManualSenderOptions {
+    pub sender_address: Option<String>,
+}
+
+/// Signer of the transactions and helper for address derivation
+/// This is the main interface for simulating and signing transactions
+#[derive(Clone)]
+pub struct ManualSender {
+    pub sender: Addr,
+    pub chain_info: ChainInfoOwned,
+    pub grpc_channel: Channel,
+}
+
+impl SenderTrait for ManualSender {
+    type Error = DaemonError;
+    type SenderOptions = ManualSenderOptions;
+
+    async fn commit_tx_any(
+        &self,
+        msgs: Vec<Any>,
+        memo: Option<&str>,
+    ) -> Result<CosmTxResponse, DaemonError> {
+        // We simulate
+        let gas_needed = self.simulate(msgs, memo).await?;
+
+        log::info!("Here is the transaction to sign and broadcast: ");
+        log::info!("Gas needed: {}", gas_needed);
+
+        log::info!("Enter the txhash to proceed");
+        let stdin = io::stdin();
+        let txhash = stdin.lock().lines().next().unwrap().unwrap();
```

Comment:
Nit: you can use read_line for getting 1 line https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_line